src/wp-includes/block-supports/background.php

```diff
@@ -67,6 +68,12 @@ function wp_render_background_support( $block_content, $block ) {
 	$background_size         = isset( $block_attributes['style']['background']['backgroundSize'] )
 		? $block_attributes['style']['background']['backgroundSize']
 		: 'cover';
+	$background_position     = isset( $block_attributes['style']['background']['backgroundPosition'] )
```

Comment:
Could we check if a background image url exists as part of the condition that checks for block support above? That way we could return early if there is no image. Or are there reasons to still go through this logic in the absence of a url?

Reply:
We could do an early return. It was mostly laid out like it currently is to support subsequent updates where we might have a range of different values for the background image source. I.e.

* `'featured_image' === $background_image_source`
* `'id' === $background_image_source` (for referencing a media library item directly), etc

So we could do an early return just before we define `$background_size`, something like:

```
if ( ! $background_image_source && ! $background_image_url ) {
  return $block_content;
}
```

Would that be worth doing, do you think?

Reply:
Wouldn't those sources still require defining a URL though? We always need the URL in order to output the styles. But yeah I guess in any case, in subsequent updates we can add to the condition if that changes.

Reply:
Okie-doke! I'll update with an early return 👍

For the `featured_image` case we won't have a `url` in the block attributes, as that would get generated at render time. But that could happen anywhere we like in the function, so we could always put that logic before this `$background_position` line and do an early return if we don't have a featured image, either. Let's take a look at that part when we go to implement the feature 🙂

Reply:
Added an early return in https://github.com/WordPress/wordpress-develop/pull/5828/commits/c2982a3b381ecf9bb29df3b1d5a7b2569d277705 — let me know if you'd like me to tweak it!